--- /Users/ieb/Apache/sling/trunk/bundles/jcr/resource/src/main/java/org/apache/sling/jcr/resource/internal/JcrResourceResolver2.java	2009-09-30 10:27:39.000000000 +0100
+++ src/main/java/org/apache/sling/jcr/resource/internal/JcrResourceResolver2.java	2009-09-30 15:34:26.000000000 +0100
@@ -234,7 +234,7 @@
             log.debug(
                 "resolve: Path {} does not resolve, returning NonExistingResource at {}",
                 absPath, realPathList[0]);
-            res = new NonExistingResource(this, ensureAbsPath(realPathList[0]));
+            res = resolveNonExistingResoruce(ensureAbsPath(realPathList[0]));
         } else {
             log.debug("resolve: Path {} resolves to Resource {}", absPath, res);
         }
@@ -242,6 +242,15 @@
         return res;
     }
 
+    /**
+     * @param jcrResourceResolver2
+     * @param string
+     * @return
+     */
+    protected Resource resolveNonExistingResoruce(String absRealPath) {
+      return new NonExistingResource(this, absRealPath);
+    }
+
     // calls map(HttpServletRequest, String) as map(null, resourcePath)
     public String map(String resourcePath) {
         return map(null, resourcePath);
--- /Users/ieb/Apache/sling/trunk/bundles/jcr/resource/src/main/java/org/apache/sling/jcr/resource/internal/JcrResourceResolverFactoryImpl.java	2009-09-30 10:27:39.000000000 +0100
+++ src/main/java/org/apache/sling/jcr/resource/internal/JcrResourceResolverFactoryImpl.java	2009-09-30 15:36:20.000000000 +0100
@@ -225,6 +225,16 @@
             session, rootProviderEntry, getJcrResourceTypeProviders(),
             this.getDynamicClassLoader());
 
+        return  getResourceResolver(sessionRoot, mapEntries);
+     }
+  
+     /**
+      * @param sessionRoot
+      * @param mapEntries2
+      * @return
+      */
+     protected ResourceResolver getResourceResolver(JcrResourceProviderEntry sessionRoot,
+         MapEntries mapEntries2) {
         return new JcrResourceResolver2(sessionRoot, this, mapEntries);
     }
 
@@ -598,6 +608,14 @@
         return repository;
     }
 
+    protected void bindRepository(SlingRepository slingRepository) {
+      this.repository = slingRepository;
+    }
+    
+    protected void unbindRepository(SlingRepository slingRepository) {
+      this.repository = null;
+    }
+
     protected static final class JcrResourceTypeProviderEntry {
         final long serviceId;
 
